src/controller.py

```diff
@@ -0,0 +1,102 @@
+"""
+A script to receive messages in the waveform queue and write them to stdout,
+based on https://www.rabbitmq.com/tutorials/tutorial-one-python
+"""
+
+import json
+from datetime import datetime
+import logging
+import pika
+import db as db  # type:ignore
+import settings as settings  # type:ignore
+import csv_writer as writer  # type:ignore
+
+logging.basicConfig(format="%(levelname)s:%(asctime)s: %(message)s")
+logger = logging.getLogger(__name__)
+
+
+emap_db = db.starDB()
+emap_db.init_query()
+emap_db.connect()
+
+
+class waveform_message:
+    def __init__(self, ch, delivery_tag, body):
+        self.ch = ch
+        self.delivery_tag = delivery_tag
+        self.body = body
+
+
+def ack_message(ch, delivery_tag):
+    """Note that `ch` must be the same pika channel instance via which the
+    message being ACKed was retrieved (AMQP protocol constraint)."""
+    if ch.is_open:
+        ch.basic_ack(delivery_tag)
+    else:
+        logger.warning("Attempting to acknowledge a message on a closed channel.")
+
+
+def reject_message(ch, delivery_tag, requeue):
+    if ch.is_open:
+        ch.basic_reject(delivery_tag, requeue)
+    else:
+        logger.warning("Attempting to not acknowledge a message on a closed channel.")
+
+
+def waveform_callback(ch, method_frame, _header_frame, body):
+    data = json.loads(body)
+    try:
+        location_string = data["mappedLocationString"]
+        observation_time = data["observationTime"]
+    except IndexError as e:
+        reject_message(ch, method_frame.delivery_tag, False)
+        logger.error(
+            f"Waveform message {method_frame.delivery_tag} is missing required data {e}."
+        )
+        return
+
+    observation_time = datetime.fromtimestamp(observation_time)
```

Comment:
```suggestion
    observation_time = datetime.fromtimestamp(observation_time, tz=timezone.utc)
```
Without the tz arg, it will convert this to a timezone-naive local time representation. We probably should run our containers in UTC anyway, but best not to rely on it...